01_Build_Process/02_Boot_Protocols.md

```diff
@@ -203,115 +203,121 @@ The interface between a higher level language like C and assembly (or another hi
-- `tags`: A pointer to the first stivale2 tag in the linked list of requests.
+The main requirement of any limine protocol variable, request is that, is that the compiler keep them as they are, without any optimization optimizing them, for this reason they are declared as `volatile` and they should be accessed at least once, or marked as `used`.
 
-In the example above we actually set the first tag to a framebuffer request, so lets see what that would look like:
+In this protocol, `requests` can be placed anywhere, and there is no need to use any type of list or place them in a specific memory area. For example, let's imagine that we want to get the framebuffer information from the bootloader. In this case we need to declare the `struct limine_framebuffer_request` type while creating a variable for our request:
 
 ```c
-static stivale2_header_tag_framebuffer framebuffer_tag =
-{
-    .tag =
-    {
-        .identifier = STIVALE2_HEADER_TAG_FRAMEBUFFER,
-        .next = 0,
-    },
-    .framebuffer_width = 0,
-    .framebuffer_height = 0,
-    .framebuffer_bpp = 0
+__attribute__((used, section(".requests")))
+static volatile struct limine_framebuffer_request {
+    .id = LIMINE_FRAMEBUFFER_REQUEST,
+    .revision = 0
 };
 ```
 
-The `framebuffer_*` fields can be used to ask for a specific kind of framebuffer, but leaving them to zero tells the bootloader we want to best possible available. The `next` field can be used to point to the next header tag, if we had another one we wanted. The full list of tags is available in the stivale2 specification (see the useful links appendix).
+The requests types are all declared in the `limine.h` header.
 
-The last detail is to change the signature of our kernel entry function to:
+Any type of information that needs to be requested to the bootloader, will have it's own `request` variable, with it's type.
+
+Finally the requests start and end markers needs to be declared:
 
 ```c
-void kernel_start(stivale2_struct* stivale2_data);
+__attribute__((used, section(".requests_start_marker")))
+static volatile LIMINE_REQUESTS_START_MARKER;
+
+__attribute__((used, section(".requests_end_marker")))
+static volatile LIMINE_REQUESTS_END_MARKER;
 ```
 
-This struct points to a list of tags, each containing details about the machine we're booted on. These are called struct tags (bootloader -> kernel) as opposed to the tags we defined before (header tags: kernel -> bootloader). To get info about a specific feature, simply walk the linked list of tags, the next tag's address is available in the `tag->next` field. The end of the list is indicated by a nullptr.
+Again they can be placed anywhere in the code, since their position will be decided by the section they belongs to.
+
+The last detail is to add the kernel start function (declared in the `ENTRY()` section in the linker script):
```

Comment:
I  think we remove this sentence too, since it referred to the block of code that is now gone :)